keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,12 +263,138 @@ tried this feature in Alpha, we would have time to fix issues.
+yes, there are unit tests which make sure the field is correctly dropped 
+on feature enable and disabled
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+MaxUnavailable can fail to rollout if the new image being deployed is bad and causes the 
+containers to crash. If there are not enough instances, it can bring down many or 
+cause disruption if the maxUnavailable is set to a large number, larger than what 
+the service can tolerate. 
+If the StatefulSet is already in a failed state with more than maxUnavailable pods 
+failing, changing maxUnavailable will not help, and it will require first fixing the 
+failing pod before the system can recover.
+
+It can impact already running workloads if service owner accidentally sets maxUnavailable
+to a very high number.
+
+###### What specific metrics should inform a rollback?
+The customer should look at the behavior of the maxUnavailable and if the rollout
+is not respecting the maxUnavailable, the feature should be rolledback.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+No
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
+a value different than 1.
+The below command should show maxUnavailable value:
+```
+kubectl get statefulsets -o yaml | grep maxUnavailable
+```
+
+###### How can someone using this feature know that it is working for their instance?
+Ensure the number of replicas in your StatefulSet is more than 2. 
+Set maxUnavailable to a value 2 or more and then trigger a rolling update. 
+PMP Parallel:
+If you see the number of pods equal to maxUnavailable getting deleted at the same time,
+this means the feature is working as expected. After maxUnavailable pods are deleted,
+you will see a burst of pods equal to the number deleted up, start to come up
+
+PMP OrderedRedy:
+If you see the number of pods equal to maxUnavailable getting deleted at the same time,
+this means the feature is working as expected. After maxUnavailable pods are deleted,
+you will see a new pod come up(without violating ordering) and when it becomes ready,
+another will come up till we have all pods as ready that were deleted.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+controller-manager 
```

Comment:
Please remove - this is somewhat obvious.

There aren't any additional dependencies (we more want to ensure that you don't depend on things like dns, additonal agents, cloud-provider, etc.)

Reply:
done